x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/interceptor/FieldAndDocumentLevelSecurityRequestInterceptor.java

```diff
@@ -50,8 +51,9 @@ public void intercept(
         if (requestInfo.getRequest()instanceof IndicesRequest indicesRequest
             && false == TransportActionProxy.isProxyAction(requestInfo.getAction())) {
             // TODO: should we check is DLS/FLS feature allowed here
-            if (supports(indicesRequest)) {
-                final boolean isDlsLicensed = DOCUMENT_LEVEL_SECURITY_FEATURE.checkWithoutTracking(licenseState);
+            final boolean isDlsLicensed = DOCUMENT_LEVEL_SECURITY_FEATURE.checkWithoutTracking(licenseState);
+            final boolean isFlsLicensed = FIELD_LEVEL_SECURITY_FEATURE.checkWithoutTracking(licenseState);
+            if ((isDlsLicensed || isFlsLicensed) && supports(indicesRequest)) {
```

Comment:
nit: here I also think there would be easier to merge these two `if` conditions in a single one.